spark/src/test/scala/com/gu/tableversions/spark/VersionContextSpec.scala

```diff
@@ -105,19 +105,25 @@ class VersionedDatasetSpec extends FlatSpec with Matchers with SparkHiveSuite {
   "Inserting a snapshot dataset" should "write the data to the versioned location and commit the new version" in {
     val usersTable = TableDefinition(TableName(schema, "users"), tableUri, PartitionSchema.snapshot, FileFormat.Orc)
 
-    // Stub metastore
     val initialTableVersion = SnapshotTableVersion(Version.Unversioned)
-
     val stubbedChanges = TableChanges(List(UpdateTableVersion(version1)))
-    implicit val stubMetastore: Metastore[IO] = new StubMetastore(
-      currentVersion = initialTableVersion,
-      computedChanges = stubbedChanges
-    )
 
-    implicit val tableVersions: TableVersions[IO] = (for {
-      t <- InMemoryTableVersions[IO]
-      _ <- t.init(usersTable.name, isSnapshot = true, UserId("test"), UpdateMessage("init"), Instant.now())
-    } yield t).unsafeRunSync()
+    val versionContext = {
+      val stubMetastore: Metastore[IO] = new StubMetastore(
+        currentVersion = initialTableVersion,
+        computedChanges = stubbedChanges
+      )
+
+      val tableVersions: TableVersions[IO] = (for {
+        t <- InMemoryTableVersions[IO]
+        _ <- t.init(usersTable.name, isSnapshot = true, UserId("test"), UpdateMessage("init"), Instant.now())
+      } yield t).unsafeRunSync()
```

Comment:
This isn't related to your changes, but this bit of code is repeated a few times. Might be nice pulling it out somewhere. Could `TestVersionContext` be used here as well maybe? It would have to move into the `spark` module to do that, of course. But maybe that would be a step too far...

Reply:
I did have a look at this, but nothing jumped out quickly.

There's a bit of me that quite likes it anyway - makes it easy to see the behaviour that we're going to observe when interacting with the `VersionContext` in the test.

Will keep an eye on it!